web-wallet/src/lib/dusk/string/cleanNumberString.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Cleans the input string
+ * Returns a valid number in string form with the correct decimal separator according to the locale
+ *
+ * @param {string} amount
+ * @param {string} separator
+ * @returns {string}
+ */
+const cleanANumberString = (amount, separator) => {
+  const regex = new RegExp(`[^\\d${separator}]+`, "g"); // Remove any character that are not digits or the decimal separator
+  const regex2 = new RegExp(`(?<=\\${separator}.*)\\${separator}`, "g"); // Remove all but the first decimal separator
+  const regex3 = new RegExp(`\\${separator}+`, "g"); //Collapse multiple decimal separator into one
```

Comment:
Don't know if it's me, but I'd rather keep the last separator rather than the first (using a look ahead instead of a look behind?).

Having said that: is the third RegExp necessary at all? Shouldn't multiple separators already stripped out by the second one?

Reply:
You're right the third one is overkill and not needed. As for which separator to keep I guess it comes down to preference.